client/settings/settings.tsx

```diff
@@ -1,228 +1,370 @@
-        />
-      )
+  let contents: React.ReactNode | undefined
+  let title: string | undefined
+  switch (subPage) {
+    case AppSettingsSubPage.Sound:
+      contents = IS_ELECTRON ? <AppSoundSettings /> : <div></div>
+      title = IS_ELECTRON ? 'Sound settings' : ''
       break
-    case SettingsTab.Input:
-      contents = (
-        <InputSettings
-          scrSettings={tempScrSettings}
-          formRef={formRef}
-          onChange={onSettingsChange}
-          onSubmit={onSettingsSubmit}
-        />
-      )
+    case AppSettingsSubPage.System:
+      contents = IS_ELECTRON ? <AppSystemSettings /> : <div></div>
+      title = IS_ELECTRON ? 'System settings' : ''
       break
-    case SettingsTab.Sound:
-      contents = (
-        <SoundSettings
-          scrSettings={tempScrSettings}
-          formRef={formRef}
-          onChange={onSettingsChange}
-          onSubmit={onSettingsSubmit}
-        />
-      )
+    case GameSettingsSubPage.StarCraft:
+      contents = IS_ELECTRON ? <StarcraftSettings /> : <div></div>
+      title = IS_ELECTRON ? 'StarCraft' : ''
       break
-    case SettingsTab.Video:
-      contents = (
-        <VideoSettings
-          scrSettings={tempScrSettings}
-          formRef={formRef}
-          onChange={onSettingsChange}
-          onSubmit={onSettingsSubmit}
-        />
-      )
+    case GameSettingsSubPage.Input:
+      contents = IS_ELECTRON ? <GameInputSettings /> : <div></div>
+      title = IS_ELECTRON ? 'Input settings' : ''
       break
-    case SettingsTab.Gameplay:
-      contents = (
-        <GameplaySettings
-          localSettings={tempLocalSettings}
-          scrSettings={tempScrSettings}
-          formRef={formRef}
-          onChange={onSettingsChange}
-          onSubmit={onSettingsSubmit}
-        />
-      )
+    case GameSettingsSubPage.Sound:
+      contents = IS_ELECTRON ? <GameSoundSettings /> : <div></div>
+      title = IS_ELECTRON ? 'Sound settings' : ''
+      break
+    case GameSettingsSubPage.Video:
+      contents = IS_ELECTRON ? <GameVideoSettings /> : <div></div>
+      title = IS_ELECTRON ? 'Video settings' : ''
+      break
+    case GameSettingsSubPage.Gameplay:
+      contents = IS_ELECTRON ? <GameplaySettings /> : <div></div>
+      title = IS_ELECTRON ? 'Gameplay settings' : ''
       break
     default:
-      contents = assertUnreachable(activeTab)
+      contents = assertUnreachable(subPage)
   }
 
   return (
-    <Dialog
-      dialogRef={dialogRef}
-      title='Settings'
-      titleAction={titleAction}
-      tabs={tabs}
-      alwaysHasTopDivider={true}
-      buttons={buttons}
-      onCancel={onCancel}>
-      <ContentsBody>
-        {lastError ? (
-          <ErrorText>There was an issue saving the settings. Please try again.</ErrorText>
+    <Container style={style}>
+      <NavContainer>
+        {IS_ELECTRON ? (
+          <>
+            <NavSectionTitle>App settings</NavSectionTitle>
+            <NavEntry
```

Comment:
This is also a mess of repeated code with zero changed logic. Write a component that returns the right text label given a subpage enum value (so that they can be translated). Use that component in this piece, as well as in the component you write for rendering the content (from the above comment).

Sections can be rendered as:
```tsx
[AppSettingsSubPage.Sound, AppSettingsSubPage.System].map(s => <NavEntry key={s} subPage={s} isActive={subPage === s} onChangeSubPage={onChangeSubPage} />)
```